commit 9f1abddb1af5b94d5e82dd12e7fa8ebf18acfc64
Author: Thierry Reding <thierry.reding@avionic-design.de>
Date:   Wed Mar 23 15:57:52 2011 +0100

    Fix "make distcheck".
    
    This commit adds some missing files (some header files, the files below
    /etc and the bison/flex files) to the distribution tarball to ensure
    that libnl can be built from the tarballs created using "make dist".
    
    It also adds some incantations to properly generate the flex and bison
    output since the generated output is no longer shipped in the tarball.

diff --git a/Makefile.am b/Makefile.am
index 5931a9e..07017c8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,6 +15,9 @@ pkgconfig_DATA = libnl-3.0.pc
 sysconfdir = @sysconfdir@/libnl
 sysconf_DATA = etc/pktloc etc/classid
 
+EXTRA_DIST = \
+	$(sysconf_DATA)
+
 .PHONY: cscope
 cscope:
 	cscope -b -q -R -Iinclude -slib -ssrc;
diff --git a/include/Makefile.am b/include/Makefile.am
index e7f1206..50fde24 100644
--- a/include/Makefile.am
+++ b/include/Makefile.am
@@ -1,6 +1,17 @@
 # -*- Makefile -*-
 
 nobase_include_HEADERS = \
+	netlink/cli/addr.h \
+	netlink/cli/class.h \
+	netlink/cli/cls.h \
+	netlink/cli/ct.h \
+	netlink/cli/link.h \
+	netlink/cli/neigh.h \
+	netlink/cli/qdisc.h \
+	netlink/cli/route.h \
+	netlink/cli/rule.h \
+	netlink/cli/tc.h \
+	netlink/cli/utils.h \
 	netlink/fib_lookup/lookup.h \
 	netlink/fib_lookup/request.h \
 	netlink/genl/ctrl.h \
@@ -28,9 +39,17 @@ nobase_include_HEADERS = \
 	netlink/netlink.h \
 	netlink/object-api.h \
 	netlink/object.h \
+	netlink/route/cls/ematch/cmp.h \
+	netlink/route/cls/ematch/meta.h \
+	netlink/route/cls/ematch/nbyte.h \
+	netlink/route/cls/ematch/text.h \
+	netlink/route/cls/basic.h \
+	netlink/route/cls/cgroup.h \
+	netlink/route/cls/ematch.h \
 	netlink/route/cls/fw.h \
 	netlink/route/cls/police.h \
 	netlink/route/cls/u32.h \
+	netlink/route/link/api.h \
 	netlink/route/link/info-api.h \
 	netlink/route/link/vlan.h \
 	netlink/route/qdisc/cbq.h \
@@ -49,6 +68,7 @@ nobase_include_HEADERS = \
 	netlink/route/neighbour.h \
 	netlink/route/neightbl.h \
 	netlink/route/nexthop.h \
+	netlink/route/pktloc.h \
 	netlink/route/qdisc.h \
 	netlink/route/route.h \
 	netlink/route/rtnl.h \
@@ -59,3 +79,33 @@ nobase_include_HEADERS = \
 	netlink/types.h \
 	netlink/utils.h \
 	netlink/version.h
+
+noinst_HEADERS = \
+	linux/fib_rules.h \
+	linux/genetlink.h \
+	linux/gen_stats.h \
+	linux/if_addr.h \
+	linux/if_arp.h \
+	linux/if_ether.h \
+	linux/if.h \
+	linux/if_link.h \
+	linux/if_vlan.h \
+	linux/inetdevice.h \
+	linux/ip_mp_alg.h \
+	linux/ipv6.h \
+	linux/neighbour.h \
+	linux/netfilter.h \
+	linux/netfilter/nfnetlink_conntrack.h \
+	linux/netfilter/nfnetlink.h \
+	linux/netfilter/nfnetlink_log.h \
+	linux/netfilter/nfnetlink_queue.h \
+	linux/netlink.h \
+	linux/pkt_cls.h \
+	linux/pkt_sched.h \
+	linux/rtnetlink.h \
+	linux/snmp.h \
+	linux/tc_ematch/tc_em_meta.h \
+	netlink-generic.h \
+	netlink-local.h \
+	netlink-tc.h \
+	netlink-types.h
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 5999113..e8045b7 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -51,7 +51,6 @@ libnl_route_la_SOURCES = \
 	route/cls/fw.c route/cls/police.c route/cls/u32.c route/cls/basic.c \
 	route/cls/cgroup.c \
 	\
-	route/cls/ematch_syntax.c route/cls/ematch_grammar.c \
 	route/cls/ematch.c \
 	route/cls/ematch/container.c route/cls/ematch/cmp.c \
 	route/cls/ematch/nbyte.c route/cls/ematch/text.c \
@@ -67,8 +66,25 @@ libnl_route_la_SOURCES = \
 	\
 	fib_lookup/lookup.c fib_lookup/request.c \
 	\
-	route/pktloc_syntax.c route/pktloc_grammar.c route/pktloc.c
+	route/pktloc.c
 
+nodist_libnl_route_la_SOURCES = \
+	route/pktloc_syntax.c route/pktloc_syntax.h \
+	route/pktloc_grammar.c route/pktloc_grammar.h \
+	route/cls/ematch_syntax.c route/cls/ematch_syntax.h \
+	route/cls/ematch_grammar.c route/cls/ematch_grammar.h
+
+BUILT_SOURCES = \
+	route/cls/ematch_grammar.c \
+	route/cls/ematch_syntax.c \
+	route/pktloc_grammar.c \
+	route/pktloc_syntax.c
+
+EXTRA_DIST = \
+	route/pktloc_grammar.l \
+	route/pktloc_syntax.y \
+	route/cls/ematch_grammar.l \
+	route/cls/ematch_syntax.y
 
 if ENABLE_CLI
 nobase_pkglib_LTLIBRARIES = \
